test/puppetlabs/puppetdb/threadpool_test.clj

```diff
@@ -32,14 +32,39 @@
 
       (with-open [gtp (gated-threadpool 1 "test-pool-%d" 1000)]
         (let [fut (future
-                    (dochan gtp worker-fn in-chan))]
+                    (dochan gtp worker-fn in-chan tu/blocked?))]
           (async/>!! in-chan {:id :a})
           (is (= [:done :a] (async/<!! out-chan)))
           (is (= 1 @counter))
           (async/close! in-chan)
           (is (not= ::timed-out (deref fut tu/default-timeout-ms ::timed-out)))
           (is (true? (future-done? fut)))))))
 
+
+  (testing "block command"
```

Comment:
No strong feelings, but if there's no notable extra cost, I might even consider pulling this out into its own top-level test so it's more clearly isolated, easier to run independently, etc. e.g.
```
(deftest dochan-blocking-and-unblocking ...)
```

Reply:
Extracted the test in a top level test, added test for unblocking, decreased the timeout for blocking to `500 ms`.